src/Compiler/Checking/TailCallChecks.fs

```diff
@@ -792,7 +792,12 @@ let CheckModuleBinding cenv (isRec: bool) (TBind _ as bind) =
                     // warn for recursive calls in TryWith/TryFinally operations
                     exprs |> Seq.iter (checkTailCall true)
                 | Expr.Op(args = exprs) -> exprs |> Seq.iter (checkTailCall insideSubBindingOrTry)
-                | Expr.Sequential(expr2 = expr2) -> checkTailCall insideSubBindingOrTry expr2
+                | Expr.Sequential(expr1 = expr1; expr2 = expr2) ->
+                    match expr1 with
+                    | Expr.Op(args = exprs; op = TOp.IntegerForLoop _) -> checkTailCall insideSubBindingOrTry expr1
+                    | _ -> ()
+
+                    checkTailCall insideSubBindingOrTry expr2
```

Comment:
Why do we only need to check `expr1` if it's a for-loop?

Reply:
Hey Brian, thanks for the review.

Doing it unconditionally led to duplicated warnings in other test scenarios ( ``Warn for calls in for and iter``) and in general I like to keep changes as small as possible in this code at this stage.